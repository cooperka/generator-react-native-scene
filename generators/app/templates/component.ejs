'use extensible';

import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';

import constants from '../../../constants';
import { lightTheme as style } from '../../../styles';

import { instrumentScene } from '../../InstrumentedScene';

/**
 * <%= description %>
 */
class <%= componentName %> extends Component {

  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };

  static route = {
    navigationBar: {
      title: '<%= sceneTitle %>',
    },
  };

  render() {
    return (
      <View style={[style.sceneContainer]}>
        <Text>Hello, <%= componentName %>!</Text>
      </View>
    );
  }

}

const connectedScene = connect()(<%= componentName %>);

export default instrumentScene(connectedScene, constants.sceneKey.<%= componentConstant %>);
