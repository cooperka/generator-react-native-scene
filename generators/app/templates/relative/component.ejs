'use extensible';

import React, { Component } from 'react';
import { View } from 'react-native';

import { withInstrumentation } from 'cribspot-core/src/components/Instrumentation';

import constants from '../../../constants';
import { lightTheme as style } from '../../../styles';

import { TextOnLight as Text } from '../../semanticUi';

/**
 * <%= description %>
 */
class <%= componentName %> extends Component {

  static propTypes = {
  };

  static mapStateToProps() {
    return {};
  }

  static mapDispatchToProps() {
    return {
      doSomething() {},
    };
  }

  static sceneKey = constants.sceneKey.<%= componentNameConstant %>;

  static navigationOptions = {
    title: '<%= sceneTitle %>',
  };

  render() {
    return (
      <View style={[style.sceneContainer]}>
        <Text>Hello, <%= componentName %>!</Text>
      </View>
    );
  }

}

export default withInstrumentation(<%= componentName %>);
