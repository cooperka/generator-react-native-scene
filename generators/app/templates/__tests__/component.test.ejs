/* eslint-disable import/no-extraneous-dependencies, global-require */

import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import <%= componentName %> from '../component';

describe('<%= componentName %>', () => {
  it('renders with default props', () => {
    const tree = renderer.create(
      <<%= componentName %> />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
