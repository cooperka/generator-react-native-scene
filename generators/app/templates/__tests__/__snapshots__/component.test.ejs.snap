exports[`<%= componentName %> renders with default props 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F0F1F1",
        "flex": 1,
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Hello, <%= componentName %>!
  </Text>
</View>
`;
